plugins/woocommerce-blocks/packages/components/text-input/style.scss

```diff
@@ -9,10 +9,10 @@
 		@include reset-typography();
 		@include font-size(regular);
 		position: absolute;
-		transform: translateY(em($gap));
-		line-height: 1;
+		transform: translateY(-50%);
+		line-height: 1.5;
 		left: em($gap-smaller + 1px);
-		top: 0;
+		top: 25px;
```

Comment:
This is a bit risky, but I couldn't use % due to having input error inside the div. 25px comes from the 50px height below.

This ensures label is always in middle.

Line height increase was needed because it would cut off certain below the line letters like y

Reply:
Interesting, I think this one is a bit risky looking at twentytwentyfive. twentytwentyfive raises the font size up significantly which means button and inputs lined up together have a big height disparity. Probably partially because we have a lot of custom CSS to handle the animated label placeholder styles. 🤔

Reply:
yea im aware, we need to move from pixels to ems for that (and all inputs height). I had that working fine locally in a branch

Reply:
@senadir I didn't see that fixing the issue with matching input heights to button heights, but keen to see your branch if you have one I can look at